src/Features/Core/Portable/ChangeSignature/AbstractChangeSignatureService.cs

```diff
@@ -62,7 +60,7 @@ public async Task<ImmutableArray<ChangeSignatureCodeAction>> GetChangeSignatureC
 
         internal ChangeSignatureResult ChangeSignature(Document document, int position, Action<string, NotificationSeverity> errorHandler, CancellationToken cancellationToken)
         {
-            var context = GetContextAsync(document, position, restrictToDeclarations: false, cancellationToken: cancellationToken).WaitAndGetResult(cancellationToken);
+            var context = GetContextAsync(document, position, restrictToDeclarations: false, cancellationToken: cancellationToken).WaitAndGetResult_CanCallOnBackground(cancellationToken);
```

Comment:
ick.  we should file bug on this.  this method should be async itself.

Reply:
Opened up an issue to track this: https://github.com/dotnet/roslyn/issues/42293

Reply:
enh... can we not do this int he PR?